pyproject.toml

```diff
@@ -53,13 +48,6 @@ black = "^23.3.0"
 ruff = "^0.0.278"
 jinja2 = "^3.1.2"
 pyyaml = "^6.0"
-tox = "^4.6.3"
-pre-commit = "^3.3.3"
```

Comment:
You have removed to much - `pre-commit`, `pytest-xdist` and `pytest-cov` should remain in place. You could also check latest versions of those on PyPI and update them (if possible).